### PR TITLE
HACK Week: Support uploading local files for downloadable products

### PR DESCRIPTION
**Pull request by itsmeichigo**
Part of #5064 

## What

This PR adds an option for uploading document files for downloadable products.

## How
- Updated the bottom sheet for selecting file source with an option for document files on device.
- Added a simple helper file `LocalFileUploader` for handling uploading local files.
- Integrated `UIDocumentPickerViewController` for picking documents on device.
- Updated `MediaAction` and `MediaStore` to support uploading files with local URL.
- Handled uploading documents after picking from device.

📝 Notes:
- There's a [runtime warning](https://github.com/woocommerce/woocommerce-ios/issues/11354) on XCode 15 when uploading files - this will be handled in a separate PR.
- The media library picker is currently displaying only images, so it's not possible to pick documents from there for now.

## Testing instructions

Prerequisite: If you are testing in a simulator, you can add files to your simulator by dragging them to the simulator window.
- Log in to a store and navigate to the Products tab.
- Select any simple product or create one.
- In the product form, select the ellipsis menu (...) > Product Settings > Enable downloadable option.
- Back to the product form, select Downloadable Files > Add File > Document on device.
- Select a document of a supported type, notice that a loading view is displayed, then a screen is displayed for updating the file name.

## Screenshots

https://github.com/woocommerce/woocommerce-ios/assets/5533851/a4e97120-b329-47d9-9adc-eb833c4af5d8

---
- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

### CODE REVIEW

File: Yosemite/Yosemite/Stores/MediaStore.swift [hunk 1]
```diff
@@ -159,6 +165,24 @@ private extension MediaStore {
         }
     }
 
+    func uploadFile(siteID: Int64,
```
Comment (selanthiraiyan) [inline Yosemite/Yosemite/Stores/MediaStore.swift:174]: nit: Consider adding a few unit tests for this new action. Considering that we are firing `uploadMedia` which already has enough unit tests maybe testing basic success and error cases should be enough.

**Review by selanthiraiyan — APPROVED** [review]
This tests as expected, and the code looks good to me. 🚀

**Comment by itsmeichigo** [comment]
@selanthiraiyan When adding the missing tests I realized that the uploaded files are removed since I'm using the same `uploadMedia` methods, which removes the temporary files after uploading. I updated the logic in 4f2c763eeecbd0d7430300fea8232bd7234f4734 and add tests in 33ddd939a9f519ce2d4d117273fc19c580e65920 - please feel free to test again to confirm that the local files are not deleted after uploaded.

**Comment by selanthiraiyan** [comment]
Great catch, Huong! And thanks for fixing it. 👍 

--- 

Not related to this PR. I think we should disable the `Done` button to prevent the user from leaving the screen before the upload is complete. 


https://github.com/woocommerce/woocommerce-ios/assets/524475/bda8e1e4-6c49-4fad-b787-3c8be7bd463f

**Comment by itsmeichigo** [comment]
> Not related to this PR. I think we should disable the Done button to prevent the user from leaving the screen before the upload is complete.

Nice catch! I've added a change in #11356 to improve this. I'll merge this PR for now.

**Pull request merged**